### ISSUE TITLE
Rigid Table Manners does not work as described

### ISSUE THREAD

**Issue by PaulBlay** [CleverRaven/Cataclysm-DDA#45240]
# Describe the bug

You get “Ate like an animal” morale penalties when you should not.

Even though I was eating inside an RV sitting on a seat next to a table I got the morale penalty.  According to the flavour text for Rigid Table Manners “Eating without a table and chair frustrates you, but eating like a civilized person gives you a bigger morale bonus.”

# Steps To Reproduce

1. Cook some food.
2. Stand on a tile with a chair next to a tile with a table.
3. Eat the food.

# Expected behavior

I expected to get a morale bonus, and not to get a morale penalty.

# Versions and configuration

<!-- Please complete the following information; you can now go to:
    - "Main menu (ESC) > Debug Menu > Info > Generate game report" to generate this report.
    - Please update your OS version with a more precise version.

- OS: Windows
    - OS Version: MINGW/CYGWIN/MSYS2 on Windows 8.1
- Game Version: 0.E-7239-gf5bd4da [64-bit]
- Graphics Version: Tiles
- Game Language: English [en]
- Mods loaded: [
    Dark Days Ahead [dda],
    Disable NPC Needs [no_npc_food],
    SpeedyDex [speedydex],
    Stats Through Skills [StatsThroughSkills],
    Stats Through Kills [stats_through_kills]
]

# Additional context

I also had ceramic plate, pewter bowl, spoon and carving fork - but I don’t think those affect anything.

**Comment by PaulBlay** [comment]
Edit doesn't work? For some reason the version information got lost when I posted it, so I'm putting it here.

- OS: Windows
    - OS Version: MINGW/CYGWIN/MSYS2 on Windows 8.1
- Game Version: 0.E-7239-gf5bd4da [64-bit]
- Graphics Version: Tiles
- Game Language: English [en]
- Mods loaded: [
    Dark Days Ahead [dda],
    Disable NPC Needs [no_npc_food],
    SpeedyDex [speedydex],
    Stats Through Skills [StatsThroughSkills],
    Stats Through Kills [stats_through_kills]
]

**Comment by kevingranade** [comment]
This issue has been mentioned on **Cataclysm: Dark Days Ahead**. There might be relevant details there:

https://discourse.cataclysmdda.org/t/rigid-table-manners-not-working-as-described-11131/25158/1

**Comment by ZhilkinSerg** [comment]
Duplicate of https://github.com/CleverRaven/Cataclysm-DDA/issues/29384

**Comment by PaulBlay** [comment]
#29384 is closed, which is why my search didn't find it. :/

**Comment by wapcaplet** [comment]
#29384 was closed because it was actually fixed, so this new issue is more of a regression than a duplicate.

**Comment by tsulh** [comment]
I fixed the bug locally this way: in the file src/map.cpp, map "has_nearby_table", line 2819, replace "veh_at( p )" with "veh_at( pt )".

However, I am having difficulty committing the fix (I don't use git for anything but cdda). I understand that I need to create a branch with my change, commit it and make a pull request.
Below are the commands I used (in a linux terminal), which resulted in an error. I'd appreciate an advice on what I did wrong, since I don't have time to investigate this issue.

$ git checkout -b bugfix_for_45240 
$ git add src/map.cc
$ git commit -m "bugfix for #45240"
$ git push
ERROR: 
fatal: The current branch bugfix_for_45240 has no upstream branch.
To push the current branch and set the remote as upstream, use git push --set-upstream origin bugfix_for_45240
$  git push --set-upstream origin bugfix_for_45240
Username for 'https://github.com': tsulh
Password for 'https://tsulh@github.com': 
remote: Permission to CleverRaven/Cataclysm-DDA.git denied to tsulh.
fatal: unable to access 'https://github.com/CleverRaven/Cataclysm-DDA.git/': The requested URL returned error: 403
